### PR TITLE
add mint60 keyboard

### PR DESCRIPTION
**Pull request by eucalyn**
I designed a new split keyboard "Mint60"

### CODE REVIEW

File: keyboards/mint60/mint60.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  #include <avr/interrupt.h>
+#endif
+#endif
+
+
+// This a shortcut to help you visually see your layout.
+// The following is an example using the Planck MIT layout
+// The first section contains all of the arguments
+// The second converts the arguments into a two-dimensional array
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/mint60/mint60.h]: Does this match any of the existing layouts? 

https://github.com/qmk/qmk_firmware/tree/master/layouts/default

File: keyboards/mint60/pro_micro.h [hunk 2]
```diff
@@ -0,0 +1,362 @@
+/*
```
Comment (drashna) [inline keyboards/mint60/pro_micro.h]: Is this file necessary/needed?

File: keyboards/mint60/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Mint60
+
+![Mint60](https://imgur.com/a/8xnTS4U)
+
+A short description of the keyboard/project
+
+Keyboard Maintainer: [Eucalyn](https://github.com/eucalyn)  [@eucalyn_](https://twitter.com/eucalyn_)
+Hardware Supported: The Mint60 PCBs, ProMicro supported  
+Hardware Availability: links to where you can find this hardware
```
Comment (drashna) [inline keyboards/mint60/readme.md]: Do you have a link for this?  

Are you running this as a group buy, or selling the PCB/kit?  

So you plan to? 

If not, then this should go in /keyboards/handwire/ instead, as the main folder is meant more for production boards.

File: keyboards/mint60/rules.mk [hunk 4]
```diff
@@ -0,0 +1,79 @@
+SRC += i2c.c \
+	   serial.c \
+		 matrix.c \
+ 	   split_util.c \
+ 	   ws2812.c
```
Comment (drashna) [inline keyboards/mint60/rules.mk]: This line isn't needed

File: keyboards/mint60/rules.mk [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+
+CUSTOM_MATRIX = yes
+USE_I2C = yes
```
Comment (drashna) [inline keyboards/mint60/rules.mk]: This line isn't needed, nor does it do anything.

File: keyboards/mint60/serial_config.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#ifndef SOFT_SERIAL_CONFIG_H
+#define SOFT_SERIAL_CONFIG_H
+
+/* Soft Serial defines */
+#define SERIAL_PIN_DDR DDRD
+#define SERIAL_PIN_PORT PORTD
+#define SERIAL_PIN_INPUT PIND
+#define SERIAL_PIN_MASK _BV(PD2)
```
Comment (drashna) [inline keyboards/mint60/serial_config.h]: If this isn't PD0, then it can't sure serial and i2c, IIRC.  And in that case, you should only use one or the other. 

Unless I'm mis-reading.

File: keyboards/mint60/config.h [hunk 7]
```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/config.h]: Don't forget to put your name here!

File: keyboards/mint60/keymaps/default/config.h [hunk 8]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/keymaps/default/config.h]: Don't forget to put your name here!

File: keyboards/mint60/keymaps/eucalyn/config.h [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/keymaps/eucalyn/config.h]: Don't forget to put your name here!

File: keyboards/mint60/keymaps/eucalyn/keymap.c [hunk 10]
```diff
@@ -0,0 +1,105 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/keymaps/eucalyn/keymap.c]: Don't forget to put your name here!

File: keyboards/mint60/mint60.c [hunk 11]
```diff
@@ -0,0 +1,43 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/mint60.c]: Don't forget your name here!

File: keyboards/mint60/mint60.h [hunk 12]
```diff
@@ -0,0 +1,59 @@
+/* Copyright 2018 REPLACE_WITH_YOUR_NAME
```
Comment (mechmerlin) [inline keyboards/mint60/mint60.h]: Don't forget your name here!

File: keyboards/mint60/config.h [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+#define MANUFACTURER    Eucalyn
+#define PRODUCT         Mint60
+#define DESCRIPTION     A row staggered split keyboard
+
+#define PREVENT_STUCK_MODIFIERS
+#define TAPPING_FORCE_HOLD
+#define TAPPING_TERM 100
+
+/* Use I2C or Serial */
+#define USE_I2C
```
[the remark this answers is not in the document]
Comment (mtei) [reply keyboards/mint60/config.h]: Hi, I think it's likely OK.

In let's split, USE_I2C mean
```
qmk_firmware/keyboards/lets_split % find . -name '*.[ch]' | xargs grep _I2C
./i2c.c:#ifdef USE_I2C
./keymaps/default/config.h:// #define USE_I2C
./matrix.c:#ifdef USE_I2C
./matrix.c:#ifdef USE_I2C
./matrix.c:    int err = i2c_master_start(SLAVE_I2C_ADDRESS + I2C_WRITE);
./matrix.c:    err = i2c_master_start(SLAVE_I2C_ADDRESS + I2C_READ);
./matrix.c:#ifdef USE_I2C
./matrix.c:#ifdef USE_I2C
./rev2/rev2.h:#ifdef USE_I2C
./serial.c:#ifndef USE_I2C
./sockets/sockets.h:#ifdef USE_I2C
./split_util.c:#ifdef USE_I2C
./split_util.c:#ifdef USE_I2C
./split_util.c:#ifdef USE_I2C
./split_util.c:    i2c_slave_init(SLAVE_I2C_ADDRESS);
./split_util.h:#define SLAVE_I2C_ADDRESS           0x32
```

But In this ming60, There is USE_I2C and USE_MATRIX_I2C
```
qmk_firmware/keyboards/mint60 % find . -name '*.[ch]' | xargs grep _I2C
./config.h:#define USE_I2C
./config.h://#define USE_MATRIX_I2C
./i2c.c:#ifdef USE_I2C
./matrix.c:#ifdef USE_MATRIX_I2C
./matrix.c:#ifdef USE_MATRIX_I2C
./matrix.c:    int err = i2c_master_start(SLAVE_I2C_ADDRESS + I2C_WRITE);
./matrix.c:    err = i2c_master_start(SLAVE_I2C_ADDRESS + I2C_READ);
./matrix.c:#ifdef USE_MATRIX_I2C
./matrix.c:#ifdef USE_MATRIX_I2C
./matrix.c:#ifdef USE_MATRIX_I2C
./mint60.h:#ifdef USE_I2C
./split_util.c:#ifdef USE_MATRIX_I2C
./split_util.c:#ifdef USE_MATRIX_I2C
./split_util.c:#ifdef USE_MATRIX_I2C
./split_util.c:    i2c_slave_init(SLAVE_I2C_ADDRESS);
./split_util.h:#define SLAVE_I2C_ADDRESS           0x32
```

File: keyboards/mint60/readme.md (hunk 3 above)
Comment (eucalyn) [reply keyboards/mint60/readme.md]: I'm planning group buy and I will open PCBs & case data in GitHub soon.
But case data is incomplete now.

File: keyboards/mint60/mint60.h (hunk 1 above)
Comment (eucalyn) [reply keyboards/mint60/mint60.h]: I can't found similar layout.

File: keyboards/mint60/serial_config.h (hunk 6 above)
Comment (eucalyn) [reply keyboards/mint60/serial_config.h]: Thanks @mtei .
You are right.

**Comment by eucalyn** [comment]
Should I change something?

**Comment by drashna** [comment]
Sorry, looks good to go! 

Thanks
